### PR TITLE
Add information on setting up HSM Node Groups for K3s

### PR DESCRIPTION
**Pull request by keopp-hpe**
#### Summary and Scope
This PR adds documentation on adding UAN nodes to the necessary HSM node groups to support K3s configuration.

### CODE REVIEW

File: docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Designating Application Nodes for K3s (Technical Preview)
+
+**WARNING**: This feature is currently a Technical Preview. Future releases will streamline these manual configuration steps. Therefore, some of these configuration options may change in future releases.
+
+## Overview
+
+Before K3s can be enabled to support UAIs on Application nodes, the Application nodes must be labeled in HSM as either `k3s_server` or `k3s_agent` nodes. The UAN Ansible K3s playbooks use these labels to configure them. Nodes labeled `k3s_server` will become K3s control-plane (master) nodes, while nodes labeled `k3s_agant` will become K3s agent (worker) nodes available to host UAIs.
```
Comment (darrenn-jackson) [inline docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md:7]: fix typo
```suggestion
Before K3s can be enabled to support UAIs on Application nodes, the Application nodes must be labeled in HSM as either `k3s_server` or `k3s_agent` nodes. The UAN Ansible K3s playbooks use these labels to configure them. Nodes labeled `k3s_server` will become K3s control-plane (master) nodes, while nodes labeled `k3s_agent` will become K3s agent (worker) nodes available to host UAIs.
```

File: docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+When UAN software is installed or upgraded using IUF and these HSM node group labels don't exist or have no members, one Application_UAN node will be labeled `k3s_server` and the remaining nodes will be labeled `k3s_agent`.  If these groups exist and are not empty, IUF will not change them.
```
Comment (darrenn-jackson) [inline docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md:9]:
```suggestion
When UAN software is installed or upgraded using IUF and these HSM node group labels do not exist or have no members, one `Application_UAN` node type will be labeled `k3s_server` and the remaining nodes will be labeled `k3s_agent`.  If these groups exist and are not empty, IUF will not change them.
```

File: docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+This procedure will display the members of a given HSM Node Group. In the context of this document, `GROUP_NAME` would contain either `k3s_server` or `k3s_agent`.
+
+```bash
+$ cray hsm groups members list $GROUP_NAME
+```
+
+### Creating the HSM Node Group
+
+If the HSM Node Group doesn't exist, this procedure may be used to create it. The group members may be provided as a comma-separated string of XNAMEs on the command line, or listed in a file.
```
Comment (darrenn-jackson) [inline docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md:36]:
```suggestion
If the HSM Node Group doesn't exist, the `cray hsm groups create` command can create it. The following two examples show how group members may be provided as a comma-separated string of XNAMEs on the command line, or listed in a file.
```

File: docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+ $ cray hsm groups create \
+ --members-file /path-to-member-file \
+ --exclusive-group "K3s" \
+ --description "K3s Agent Nodes"\
+ --label "k3s_agent"
+   ```
+
+### Modifying the HSM Node Group
+
+This procedure is used to modify membership of an existing HSM Node Group. In these examples, GROUP_NAME would be either `k3s_server` or `k3s_agent`.
```
Comment (darrenn-jackson) [inline docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md:62]:
```suggestion
The `cray hsm groups members create` and `cray hsm groups members delete` commands modify membership of an existing HSM Node Group. In these examples, `GROUP_NAME` would be either `k3s_server` or `k3s_agent`.
```

File: docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+#### Deleteing a Member
+
+```bash
+$ cray hsm groups members delete XNAME $GROUP_NAME
+```
+
+### Deleting a HSM Node Group
+
+This procedure will delete the node group contained in GROUP_NAME.
```
Comment (darrenn-jackson) [inline docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md:78]:
```suggestion
This command deletes the node group contained in `GROUP_NAME`.
```

**Review by darrenn-jackson — CHANGES_REQUESTED** [review]
A bunch of small improvements.

File: docs/portal/developer-portal/hugo_toc.json [hunk 6]
```diff
@@ -118,9 +118,13 @@
     "weight": 54
   },
   {
-    "title": "Configuring a UAN for K3s (Technical Preview)",
+    "title": "Designating Application Nodes for K3s (Technical Preview)",
     "weight": 55
   },
+  {
+    "title": "Configuring a UAN for K3s (Technical Preview)",
```
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply docs/portal/developer-portal/hugo_toc.json:126]: There was a miss on my part in adding the new file further up the list.  I pushed a new commit to fix it.

**Review by darrenn-jackson — APPROVED** [review]

File: docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+# Designating Application Nodes for K3s (Technical Preview)
+
+**WARNING**: This feature is currently a Technical Preview. Future releases will streamline these manual configuration steps. Therefore, some of these configuration options may change in future releases.
+
+## Overview
+
+Before K3s can be enabled to support UAIs on Application nodes, the Application nodes must be labeled in HSM as either `k3s_server` or `k3s_agent` nodes. The UAN Ansible K3s playbooks use these labels to configure them. Nodes labeled `k3s_server` will become K3s control-plane (master) nodes, while nodes labeled `k3s_agent` will become K3s agent (worker) nodes available to host UAIs.
```
Comment (alanm-hpe) [inline docs/portal/developer-portal/advanced/Designating_Application_Nodes_for_K3s.md:7]:
```suggestion
Before K3s can be enabled to support UAIs on Application nodes, the Application nodes must be grouped in HSM as either `k3s_server` or `k3s_agent` nodes. The UAN Ansible K3s playbook use these groups to determine what role they will have. Nodes grouped as `k3s_server` will become K3s control-plane (master) nodes, while nodes grouped as `k3s_agent` will become K3s agent (worker) nodes.
```

**Comment by alanm-hpe** [comment]
Could you add a link to https://github.com/Cray-HPE/docs-csm/blob/release/1.6/operations/configuration_management/Ansible_Inventory.md? It provides some useful information on ansible dynamic inventory and hsm groups.

**Comment by keopp-hpe** [comment]
> Could you add a link to https://github.com/Cray-HPE/docs-csm/blob/release/1.6/operations/configuration_management/Ansible_Inventory.md? It provides some useful information on ansible dynamic inventory and hsm groups.
> 
> Not an actual link, but a reference to the "configuration_management/Ansible_Inventory" topic.

I added this:

**INFO:** For more information on how CFS uses HSM node groups to create Ansible host groups, see the [Cray System Management Documentation](https://cray-hpe.github.io/docs-csm). Follow the links to the `Cray System Management Administration Guide->Configuration Management->Ansible Inventory->Dynamic inventory and host groups` section.

**Review by alanm-hpe — APPROVED** [review]

**Pull request merged**